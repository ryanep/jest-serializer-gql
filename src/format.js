import { buildASTSchema, printSchema } from 'graphql';

export default (schema, options = {}) => {
  if (options.ast) {
    return JSON.stringify(schema, null, 2);
  }

  const ast = buildASTSchema(schema);
  return printSchema(ast).trim();
};
