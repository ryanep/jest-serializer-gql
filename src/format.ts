import { ASTNode } from 'graphql';
import { print } from 'graphql/language/printer';

export const format = (node: ASTNode): string => {
  const output = print(node).trim();
  return output;
};
