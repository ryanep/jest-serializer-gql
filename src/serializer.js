import createSerializer from './create-serializer';

export default createSerializer();
