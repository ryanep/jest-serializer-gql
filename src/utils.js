export const isGraphqlSchema = schema => {
  if (
    typeof schema === 'object' &&
    schema.kind &&
    schema.definitions &&
    schema.loc
  ) {
    return true;
  }
  return false;
};
