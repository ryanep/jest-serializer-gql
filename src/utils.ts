import { ASTNode } from 'graphql';

export const isGraphqlSchema = (node: ASTNode): boolean => {
  if (typeof node === 'object' && node.kind && node.loc) {
    return true;
  }
  return false;
};
