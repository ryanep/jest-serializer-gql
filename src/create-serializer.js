import { isGraphqlSchema } from './utils';
import format from './format';

export default function createSerializer(options) {
  return {
    test(val) {
      return isGraphqlSchema(val);
    },
    print(val) {
      return format(val, options);
    },
  };
}
