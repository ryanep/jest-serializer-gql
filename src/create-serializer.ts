import { ASTNode } from 'graphql';
import { isGraphqlSchema } from './utils';
import { format } from './format';

export const createSerializer = (): jest.SnapshotSerializerPlugin => {
  return {
    test(node: ASTNode): boolean {
      return isGraphqlSchema(node);
    },
    print(node: ASTNode): string {
      return format(node);
    },
  };
};
