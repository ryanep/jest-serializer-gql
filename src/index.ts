import { createSerializer } from './create-serializer';

export { createSerializer };
