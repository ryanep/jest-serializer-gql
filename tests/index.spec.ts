import gql from 'graphql-tag';
import { createSerializer } from '../src';

const schema = `
	type User {
    id: ID!
    firstName: String! @deprecated
	}

	type Query {
		users(firstName: String): [User]
	}

	type Mutation {
		login(emailAddress: String! password: String!): User
	}
`;

const query = `
  query users {
    users {
      id
    }
  }
`;

describe('graphql schema', (): void => {
  it('should match graphql schema snapshot', (): void => {
    expect.addSnapshotSerializer(createSerializer());
    const ast = gql(schema);
    expect(ast).toMatchSnapshot();
  });

  it('should match graphql query snapshot', (): void => {
    expect.addSnapshotSerializer(createSerializer());
    const ast = gql(query);
    expect(ast).toMatchSnapshot();
  });
});
