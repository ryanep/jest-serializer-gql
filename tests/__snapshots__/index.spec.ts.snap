// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql schema should match graphql query snapshot 1`] = `
query users {
  users {
    id
  }
}
`;

exports[`graphql schema should match graphql schema snapshot 1`] = `
type User {
  id: ID!
  firstName: String! @deprecated
}

type Query {
  users(firstName: String): [User]
}

type Mutation {
  login(emailAddress: String!, password: String!): User
}
`;
